test/lib/tilex/tracking_test.exs

```diff
@@ -31,10 +31,22 @@ defmodule Tilex.TrackingTest do
       start_date = DateTime.utc_now() |> Timex.beginning_of_week()
       end_date = DateTime.utc_now() |> Timex.end_of_week()
 
+      %{name: channel} = Factory.insert!(:channel, name: "phoenix")
```

Comment:
@avogel3 this might be a bit out of the scope of this change, but I can't see where this new `channel` is connected to the new `posts` that are being created...

https://github.com/hashrocket/tilex/pull/818/files#diff-b6892df66713f3ecdac4e0ad8da4d7ed747780d61253b001613cc56f934ad7f1R9-R27

I think that function is doing too much, OMHO creating the `:post` and `Request` are totally separated things, I'd split that func as it carries too much responsabillity. And for the post creation I'd set the `channel_id` as you just created.

Last thing, I'd expect that slug creation to be a function in the changeset, so I'd just call that generate slug when creating that post, or even better (but way more changes) would be to change the Factory to use `changeset` functions. Thoughts on that?